OasisDEX/oasis-borrow
# Add DAI Saving Rate

# [Add DAI Saving Rate](https://app.shortcut.com/oazo-apps/epic/6883/dai-savings-rate-strategy?cf_workflow=500000053&ct_workflow=all)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- added dai saving rate earn product
  
## How to test 🧪
  <Please explain how to test your changes>

- enable `DaiSavingsRate` feature toggle
- using hardhat, use DSR product card in earn tab on landing page (when connected you should be moved straight into manage dsr position view, if not connected connect view should pop up first)
- try to deposit / withdraw
- proxy & allowance should work (test all allowance options)
- if dsr net value is above 0, position should be displayed in owner page in earn category
- check simulation calculations
- check history
- try to deposit & withdraw max
- you should be able to visit other people positions by using their wallet address in url
- validation should pop up when you try to deposit more DAI than you have in wallet or withdraw more than its currently in position
- if you are on someone's else position button should be disabled
- for now FAQ tab contains info from aave but we will change it before release
